src/main/java/no/digipost/collection/NonEmptyList.java

```diff
@@ -190,7 +191,22 @@ static <E> NonEmptyList<E> ofUnsafe(E[] nonEmptyArray) {
      *         or {@link Optional#empty()} if the given list is empty
      */
     static <E> Optional<NonEmptyList<E>> copyOf(List<E> list) {
-        return firstOf(list).map(first -> NonEmptyList.of(first, new ArrayList<>(list.subList(1, list.size()))));
+        return copyOf((Collection<E>) list);
+    }
+
+
+    /**
+     * Try to construct a non-empty list from copying the elements
+     * of a given collection, which may be empty.
+     *
+     * @param <E> the type of elements in the collection.
+     * @param collection the collection, which may be empty
+     *
+     * @return the resulting non-empty list,
+     *         or {@link Optional#empty()} if the given collection is empty
+     */
+    static <E> Optional<NonEmptyList<E>> copyOf(Collection<E> collection) {
+        return of(new ArrayList<>(collection));
```

Comment:
Using copy constructor from the JDK instead of the oh-so-fancy `firstOf(list).map(...)` previously used for `copyOf(List)`.